### PR TITLE
Bug 1415619: Add seperate routes for comm-central derived branches.

### PR DESCRIPTION
**Pull request by tomprince**
https://bugzilla.mozilla.org/show_bug.cgi?id=1415619

### CODE REVIEW

File: src/make-gecko-branch-role.js [hunk 1]
```diff
@@ -42,30 +49,63 @@ module.exports.run = async function(projectsOption, options) {
...
       'queue:route:tc-treeherder-stage.<project>.*',
       'queue:route:tc-treeherder.<project>.*',
       'queue:route:tc-treeherder-stage.v2.<project>.*',
       'queue:route:tc-treeherder.v2.<project>.*',
       'queue:route:coalesce.v1.builds.<project>.*',  // deprecated - https://bugzilla.mozilla.org/show_bug.cgi?id=1382204
       'queue:route:coalesce.v1.<project>.*',
       'queue:route:index.releases.v1.<project>.*',
-      'project:releng:buildbot-bridge:builder-name:release-<project>-*',
-      'project:releng:buildbot-bridge:builder-name:release-<project>_*',
+      'secrets:get:project/releng/<trust-domain>/build/level-3/*'
```
Comment (djmitche) [inline src/make-gecko-branch-role.js]: `<level>`?

**Review by djmitche — APPROVED** [review]
What does the `-n --all` output for this show?

**Review by djmitche — CHANGES_REQUESTED** [review]
oops

**Comment by tomprince** [comment]
Changes: https://gist.github.com/tomprince/fb0861596430c66e5b24f9962a80aaf5

**Comment by tomprince** [comment]
Once this is run, the following scopes should be removed from moz-tree:level:3:
- `index:insert-task:docker.images.v2.level-3.*`                                                                                                                                                                   
- `index:insert-task:gecko.cache.level-3.*`
- `queue:route:index.docker.images.v2.level-3.*`                                                                                                                                                                   
- `queue:route:index.gecko.cache.level-3.*`                                                                                                                                                                        
-  `secrets:get:project/releng/gecko/build/level-3/*`
as they are granted to the individual branches now.

**Comment by tomprince** [comment]
Also, `secrets:get:project/taskcluster/gecko/build/level-3/*` appear to not refer to any secrets.

**Review by djmitche — APPROVED** [review]

**Review by djmitche — APPROVED** [review]
Here's the diff:
```
scope changes required for role repo:hg.mozilla.org/incubator/stylo:*:
- queue:route:index.docker.images.v1.stylo.*
- index:insert-task:docker.images.v1.stylo.*
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
scope changes required for role repo:hg.mozilla.org/integration/autoland:*:
- queue:route:index.docker.images.v1.autoland.*
- index:insert-task:docker.images.v1.autoland.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/integration/autoland:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/jamun:*:
- queue:route:index.docker.images.v1.jamun.*
- index:insert-task:docker.images.v1.jamun.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/projects/jamun:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/gum:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/gum:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/fig:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/fig:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/maple:*:
- queue:route:index.docker.images.v1.maple.*
- index:insert-task:docker.images.v1.maple.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/projects/maple:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/larch:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/larch:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/try-comm-central:*:
- queue:route:index.buildbot.branches.try-comm-central.*
- index:insert-task:buildbot.branches.try-comm-central.*
- queue:route:index.buildbot.revisions.*
- index:insert-task:buildbot.revisions.*
- queue:route:index.docker.images.v1.try-comm-central.*
- index:insert-task:docker.images.v1.try-comm-central.*
- queue:route:index.gecko.v2.try-comm-central.*
- index:insert-task:gecko.v2.try-comm-central.*
- project:releng:buildbot-bridge:builder-name:release-try-comm-central-*
- project:releng:buildbot-bridge:builder-name:release-try-comm-central_*
+ queue:route:index.comm.v2.try-comm-central.*
+ index:insert-task:comm.v2.try-comm-central.*
+ index:route:index:comm.cache.level-1.*
+ index:insert-task:comm.cache.level-1.*
+ secrets:get:project/releng/comm/build/level-1/*
scope changes required for role repo:hg.mozilla.org/incubator/stylo-try:*:
+ index:route:index:gecko.cache.level-1.*
+ index:insert-task:gecko.cache.level-1.*
+ secrets:get:project/releng/gecko/build/level-1/*
+ queue:route:index.docker.images.v2.level-1.*
scope changes required for role repo:hg.mozilla.org/try:*:
+ index:route:index:gecko.cache.level-1.*
+ index:insert-task:gecko.cache.level-1.*
+ secrets:get:project/releng/gecko/build/level-1/*
+ queue:route:index.docker.images.v2.level-1.*
scope changes required for role repo:hg.mozilla.org/projects/graphics:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/graphics:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/elm:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/elm:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/ux:*:
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
scope changes required for role repo:hg.mozilla.org/projects/cypress:*:
- queue:route:index.buildbot.branches.cypress.*
- index:insert-task:buildbot.branches.cypress.*
- queue:route:index.buildbot.revisions.*
- index:insert-task:buildbot.revisions.*
- queue:route:index.docker.images.v1.cypress.*
- index:insert-task:docker.images.v1.cypress.*
- queue:route:index.gecko.v2.cypress.*
- index:insert-task:gecko.v2.cypress.*
- project:releng:buildbot-bridge:builder-name:release-cypress-*
- project:releng:buildbot-bridge:builder-name:release-cypress_*
+ queue:route:index.comm.v2.cypress.*
+ index:insert-task:comm.v2.cypress.*
+ index:route:index:comm.cache.level-2.*
+ index:insert-task:comm.cache.level-2.*
+ secrets:get:project/releng/comm/build/level-2/*
no changes required for role repo:hg.mozilla.org/projects/cypress:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/birch:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/birch:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/holly:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/holly:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/cedar:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/cedar:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/alder:*:
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/alder:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/comm-central:*:
- queue:route:index.buildbot.branches.comm-central.*
- index:insert-task:buildbot.branches.comm-central.*
- queue:route:index.buildbot.revisions.*
- index:insert-task:buildbot.revisions.*
- queue:route:index.docker.images.v1.comm-central.*
- index:insert-task:docker.images.v1.comm-central.*
- queue:route:index.gecko.v2.comm-central.*
- index:insert-task:gecko.v2.comm-central.*
- project:releng:buildbot-bridge:builder-name:release-comm-central-*
- project:releng:buildbot-bridge:builder-name:release-comm-central_*
+ queue:route:index.comm.v2.comm-central.*
+ index:insert-task:comm.v2.comm-central.*
+ index:route:index:comm.cache.level-3.*
+ index:insert-task:comm.cache.level-3.*
+ secrets:get:project/releng/comm/build/level-3/*
scope changes required for role repo:hg.mozilla.org/mozilla-central:*:
- queue:route:index.docker.images.v1.mozilla-central.*
- index:insert-task:docker.images.v1.mozilla-central.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/mozilla-central:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/integration/mozilla-inbound:*:
- queue:route:index.docker.images.v1.mozilla-inbound.*
- index:insert-task:docker.images.v1.mozilla-inbound.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/integration/mozilla-inbound:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/releases/comm-beta:*:
- queue:route:index.buildbot.branches.comm-beta.*
- index:insert-task:buildbot.branches.comm-beta.*
- queue:route:index.buildbot.revisions.*
- index:insert-task:buildbot.revisions.*
- queue:route:index.docker.images.v1.comm-beta.*
- index:insert-task:docker.images.v1.comm-beta.*
- queue:route:index.gecko.v2.comm-beta.*
- index:insert-task:gecko.v2.comm-beta.*
- project:releng:buildbot-bridge:builder-name:release-comm-beta-*
- project:releng:buildbot-bridge:builder-name:release-comm-beta_*
+ queue:route:index.comm.v2.comm-beta.*
+ index:insert-task:comm.v2.comm-beta.*
+ index:route:index:comm.cache.level-3.*
+ index:insert-task:comm.cache.level-3.*
+ secrets:get:project/releng/comm/build/level-3/*
scope changes required for role repo:hg.mozilla.org/releases/mozilla-beta:*:
- queue:route:index.docker.images.v1.mozilla-beta.*
- index:insert-task:docker.images.v1.mozilla-beta.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
+ queue:route:index.docker.images.v2.level-3.*
no changes required for role repo:hg.mozilla.org/releases/mozilla-beta:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/pine:*:
- queue:route:index.docker.images.v1.pine.*
- index:insert-task:docker.images.v1.pine.*
+ index:route:index:gecko.cache.level-2.*
+ index:insert-task:gecko.cache.level-2.*
+ secrets:get:project/releng/gecko/build/level-2/*
+ queue:route:index.docker.images.v2.level-2.*
no changes required for role repo:hg.mozilla.org/projects/pine:cron:nightly-*
scope changes required for role repo:hg.mozilla.org/projects/oak:*:
- queue:route:index.docker.images.v1.oak.*
- index:insert-task:docker.images.v1.oak.*
+ index:route:index:gecko.cache.level-3.*
+ index:insert-task:gecko.cache.level-3.*
+ secrets:get:project/releng/gecko/build/level-3/*
```

I'm a little concerned that removing those docker v1 scopes will cause issues.  I see sometihng from [16 days ago](https://tools.taskcluster.net/index/docker.images.v1.mozilla-release.desktop-build/latest) for mozilla-release, for example.

Also, you mention that the secret scopes are unused, but it looks like this patch *adds* those scopes.  I think those were added as a place to hopefully add future secrets.  I guess the idea is to move them from moz-tree:level:N to the repo roles?

**Comment by tomprince** [comment]
The releng secrets are used, the taskcluster ones aren't. I didn't move the taskcluster one because it appears unused currently.

**Comment by djmitche** [comment]
Ah, OK.

**Pull request merged**